virtualizarr/readers/hdf/hdf.py

```diff
@@ -403,3 +407,17 @@ def _get_group_attrs(
             g = f
         attrs = HDFVirtualBackend._extract_attrs(g)
         return attrs
+
+    @staticmethod
+    def _find_non_coord_dimension_vars(group: H5Group) -> List[str]:
+        dimension_names = []
+        non_coordinate_dimesion_variables = []
+        for name, obj in group.items():
+            if "_Netcdf4Dimid" in obj.attrs:
+                dimension_names.append(name)
+            for name, obj in group.items():
+                if type(obj) is h5py.Dataset:
+                    if obj.id.get_storage_size() == 0 and name in dimension_names:
+                        non_coordinate_dimesion_variables.append(name)
+
+        return non_coordinate_dimesion_variables
```

Comment:
Could this be simplified to:

```suggestion
        return [
            name
            for name, obj in group.items()
            if isinstance(obj, h5py.Dataset)
            and "_Netcdf4Dimid" in obj.attrs
            and obj.id.get_storage_size() == 0
        ]
```

Reply:
I think so, but it's melting my brain to look at

Reply:
I tried this swap locally and it worked fine.

Reply:
This was actually an indent error on my part 🤦 .  We need to first scan all items to determine all dimensions and then subsequently check if any of the those dimensions are represented by 0 size HDF5 datasets.

Reply:
You are iterating over `group.items` both times though right? So you could collapse that into one iteration. There is no way to get a name in `dimension_names` without the related `obj` having `obj.attrs == "_Netcdf4Dimid"`. 

But this is all just a nitpick. I think the only part that matters is using `isinstance` rather than `type`